### PR TITLE
The Council of bikeshed 2 electric boogaloo (LFAPI bugs solution)

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Legacy Fabric API is broken. Like hell. 
Notable features of this pull request:-
- [x] Repackagaing
- [x] Folder things
- [x] Testmods
- [x] Few more modules
- [x] Up to date stuff

## Why
The current implementation is heavily incomplete and outdated. Several parts of the API are either broken, have bugs, or are unimplemented. Some modules depend on features that don't even exist yet. The item content registry api requires a model api that loads modded item models and that feature doesn't even exist yet. Same for the armor api. 
Some modules do not match upstream. Notably, resource loader. This pull request aims to fix that. The content registry api we have is 1. misleading and 2. broken. All modules target a specific side (or don't require sides at all). No modules that add features that are required on both sides are added. 

All mods are client only or server only right now. Migration will be easy as heck.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -314,11 +311,13 @@ dependencies {
...
 	if (project.hasProperty('curse_api_key')) {
 		apiKey = project.getProperty('curse_api_key')
+	} else {
+		apiKey = ENV.CURSE_API_KEY
 	}
 	project {
 		id = '400281'
 		changelog = 'A changelog can be found at https://github.com/Legacy-Fabric/fabric/commits'
-		releaseType = 'beta'
+		releaseType = 'alpha'
```
Comment (phillyfox) [inline build.gradle:320]: Why this change?

File: modules/base/src/main/java/net/legacyfabric/fabric/api/util/TriState.java [hunk 2]
```diff
@@ -15,85 +15,31 @@
...
-		public TriState or(TriState other) {
-			return TRUE;
-		}
-
-		@Override
-		public TriState negate() {
-			return FALSE;
-		}
-	},
+	FALSE,
```
Comment (phillyfox) [inline modules/base/src/main/java/net/legacyfabric/fabric/api/util/TriState.java:34]: WHy was it like that in the first place lol

File: modules/client/rendering/build.gradle [hunk 3]
```diff
@@ -0,0 +1,5 @@
+version = getSubprojectVersion(project, "1.0.0")
+
+dependencies {
+	implementation project(path: ":legacy-fabric-api-base", configuration: "dev")
```
Comment (phillyfox) [inline modules/client/rendering/build.gradle:4]: This should use `moduleDependencies`
Ditto for all other subprojects

File: modules/client/rendering/src/main/java/net/legacyfabric/fabric/api/client/rendering/v1/BlockEntityRendererRegistry.java [hunk 4]
```diff
@@ -14,29 +15,30 @@
...
-	 * @param blockEntityType the {@link BlockEntityType} to register a renderer for
-	 * @param blockEntityRenderer a function that returns a {@link BlockEntityRenderer}
+	 * @param clazz the block entity {@link Class} to register a renderer for
+	 * @param blockEntityRenderer a function that returns a {@link BlockEntityRenderer}, called
+	 *                            when {@link BlockEntityRenderDispatcher} is initialized or immediately if the dispatcher
+	 *                            class is already loaded
 	 * @param <E> the {@link BlockEntity}
 	 */
-	<E extends BlockEntity> void register(BlockEntityType<E> blockEntityType, Function<BlockEntityRenderDispatcher, BlockEntityRenderer<E>> blockEntityRenderer);
+	<E extends BlockEntity> void register(Class<E> clazz, Function<BlockEntityRenderDispatcher, BlockEntityRenderer<E>> blockEntityRenderer);
```
Comment (phillyfox) [inline modules/client/rendering/src/main/java/net/legacyfabric/fabric/api/client/rendering/v1/BlockEntityRendererRegistry.java:43]:
```suggestion
	<E extends BlockEntity> void register(Class<E> blockEntityClass, Function<BlockEntityRenderDispatcher, BlockEntityRenderer<E>> blockEntityRenderer);
```

**Review by phillyfox — COMMENTED** [review]
Some things that I noticed

**Comment by BoogieMonster1O1** [comment]
Added tests for language loading, keybindings and feature rendering

File: modules/client/rendering/build.gradle (hunk 3 above)
Comment (BoogieMonster1O1) [reply modules/client/rendering/build.gradle:4]: next pr

File: modules/base/src/main/java/net/legacyfabric/fabric/api/util/TriState.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply modules/base/src/main/java/net/legacyfabric/fabric/api/util/TriState.java:34]: it was sponge's tristate originally

File: build.gradle (hunk 1 above)
Comment (BoogieMonster1O1) [reply build.gradle:310]: because you're missing item apis? lol

**Review by hYdos — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - uses: gradle/wrapper-validation-action@v1
-      - run: ./gradlew publish --stacktrace
+      - run: ./gradlew curseforge publish --stacktrace
         env:
-          CURSEFORGE_API_KEY: ${{ secrets.CURSEFORGE_API_KEY }}
+          CURSE_API_KEY: ${{ secrets.CURSEFORGE_API_KEY }}
```
Comment (hYdos) [inline .github/workflows/publish.yml:14]: someone remind me to add the new api key

Comment (BoogieMonster1O1) [reply .github/workflows/publish.yml:14]: bruh just add me to the org conc

Comment (BluCobalt) [reply .github/workflows/publish.yml:14]: did i not already add it

Comment (BoogieMonster1O1) [reply .github/workflows/publish.yml:14]: it’s fixed now

**Review by BluCobalt — APPROVED** [review]
please do this in small prs next time this is too big for me to comprehend conc

**Comment by BoogieMonster1O1** [comment]
> please do this in small prs next time this is too big for me to comprehend conc

Exactly why it was a big pr
